balancer-js/src/modules/data/gyro-config/repository.ts

```diff
@@ -0,0 +1,63 @@
+import { formatBytes32String } from '@ethersproject/strings';
+import { keccak256 } from '@ethersproject/solidity';
+import { defaultAbiCoder } from '@ethersproject/abi';
+import { formatFixed } from '@ethersproject/bignumber';
+import { Provider } from '@ethersproject/providers';
+import { GyroConfig, GyroConfig__factory } from '@/contracts';
+
+export interface GyroConfigRepository {
+  getGyroProtocolFee(poolAddress: string): Promise<number>;
+}
+
+export class GyroConfigRepositoryImpl implements GyroConfigRepository {
+  gyroConfig: GyroConfig;
+
+  constructor(private gyroConfigAddress: string, provider: Provider) {
+    this.gyroConfig = GyroConfig__factory.connect(gyroConfigAddress, provider);
+  }
+
+  async getGyroProtocolFee(poolAddress: string): Promise<number> {
+    let fee = 0;
+    const protocolFeePercKey = formatBytes32String('PROTOCOL_SWAP_FEE_PERC');
```

Comment:
Some of these are constant and could be moved so we don't have to recreate them each time function is called?